### DISCUSSION TITLE
How does ApplePayParams work?

### DISCUSSION THREAD

**Discussion by PeterMX** [flutter-stripe/flutter_stripe#1576] [Q&A]
**Describe the bug**
Total amount is just a single item, the last one in the cartItems list
<img src='https://apigow.urieldev.com/assets/gh_issue_34.png' width='400'>

**To Reproduce**
1. Add the next ApplePayCartSummaryItem items in cartItems from ApplePayParams
```
cartItems: [
    ApplePayCartSummaryItem.immediate(
        label: 'Producto sin modificadores',
        amount: '20.00',
    ),
    ApplePayCartSummaryItem.immediate(
        label: 'Poke',
        amount: '10.00',
    ),
    ApplePayCartSummaryItem.immediate(
        label: 'The Hawaiian',
        amount: '8.00',
    ),
]
```

**Expected behavior**
Total should be €38.00 and summary should be all three items

**Smartphone / tablet**
 - Device: iPhone 15 Pro Max
 - OS: iOS 17.0
 - Package version: 10.0.0
 - Flutter version: 3.16.5

**Additional context**
should I add a last one with the total? like:
```
cartItems: [
    ApplePayCartSummaryItem.immediate(
        label: 'Producto sin modificadores',
        amount: '20.00',
    ),
    ApplePayCartSummaryItem.immediate(
        label: 'Poke',
        amount: '10.00',
    ),
    ApplePayCartSummaryItem.immediate(
        label: 'The Hawaiian',
        amount: '8.00',
    ),
    ApplePayCartSummaryItem.immediate(
        label: 'Shopping Cart',
        amount: '38.00',
    ),
]
```

**Comment by remonh87** [comment]
This is intended behavior. The apple pay summar item has no logic displaying total etc. See [official apple docs](https://developer.apple.com/documentation/passkit/pkpaymentsummaryitem). It is up to you to name them properly and calculate the amounts.
